target/linux/ramips/dts/DWR-512-B.dts

```diff
@@ -0,0 +1,115 @@
+/dts-v1/;
+
+/include/ "rt5350.dtsi"
+
+/ {
+	compatible = "ralink,rt5350-soc";
+	model = "D-Link DWR-512 B";
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 0 1>;
+			linux,code = <0x211>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		sms {
+			label = "dwr-512-b:green:sms";
+			gpios = <&gpio0 8 1>;
+		};
+		status {
+			label = "dwr-512-b:green:status";
+			gpios = <&gpio0 9 1>;
+		};
+		2g {
+			label = "dwr-512-b:green:2g";
+			gpios = <&gpio0 17 1>;
+		};
+		3g {
+			label = "dwr-512-b:green:3g";
+			gpios = <&gpio0 19 1>;
+		};
+		sstrengthr {
+			label = "dwr-512-b:red:sigstrength";
+			gpios = <&gpio0 20 1>;
+		};
+		sstrengthg {
+			label = "dwr-512-b:green:sigstrength";
+			gpios = <&gpio0 21 1>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+ 
+		modem3g_enable {
+			gpio-export,name = "modem3g_enable";
+			gpio-export,output = <1>;
+			gpios = <&gpio0 18 1>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	mx25l6405d@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "macronix,mx25l6405d", "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <30000000>;
+		fast-read;
+
+                partition@0 {
+                        label = "u-boot";
+                        reg = <0x0 0x30000>;
+                        read-only;
+                };
+
+                partition@30000 {
+                        label = "u-boot-env";
+                        reg = <0x30000 0x10000>;
+                        read-only;
+                };
+
+                factory: partition@40000 {
+                        label = "factory";
+                        reg = <0x40000 0x10000>;
+                        read-only;
+                };
+
+                partition@50000 {
+                        label = "firmware";
+                        reg = <0x50000 0x7a0000>;
+                };
+
+		config: partition@7f0000 {
+			label = "config";
+			reg = <0x7f0000 0x10000>;
+		};
+	};
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "jtag", "uartf";
+			ralink,function = "gpio";
+		};
+	};
+};
+
+&wmac {
+	ralink,mtd-eeprom = <&config 0xe07a>;
```

Comment:
Mhh still not what I would expect to see. I have EEPROM dumps of a RT3062, RT3883 and RT5392 here. All EEPROMs have a size of 512 bytes.

The RT3062 EEPROM is from a lantiq board, which stores the EEPROM in wrong byteorder in flash. That's why it starts with 3062 instead of 6230 and the mac address is swabed as well (0c0030430062 => 00:0C:43:30:62:00). 

```
root@lede:/# hexdump -C -s 0x410 -n 512 -v /dev/mtd0 
00000410  30 62 00 01 0c 00 30 43  00 62 30 62 18 14 80 01  |0b....0C.b0b....|
00000420  00 00 30 62 18 14 00 00  00 01 ff 6a 00 0c ff ff  |..0b.......j....|
00000430  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000440  ff ff ff ff 08 22 00 26  ff ff 01 32 ff ff fa d9  |.....".&...2....|
00000450  88 cc ff ff ff 0d 00 00  00 03 00 00 00 00 ff ff  |................|
00000460  ff ff 13 13 13 13 13 13  13 13 13 13 13 13 14 14  |................|
00000470  13 13 13 13 13 13 12 12  12 12 12 12 13 13 12 14  |................|
00000480  0e 10 0a 0d 06 08 01 05  ff ff ff ff ff ff ff ff  |................|
00000490  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000004a0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000004b0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000004c0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000004d0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000004e0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff 66 66  |..............ff|
000004f0  88 88 66 88 88 88 66 88  88 88 66 88 88 88 66 88  |..f...f...f...f.|
00000500  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000510  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000520  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000530  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000540  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000550  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000560  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000570  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000580  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000590  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000005a0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000005b0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000005c0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000005d0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000005e0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000005f0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000600  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000610
```

```
$ hexdump -v -n 512 -C './mtddump/mtd2'
00000000  83 38 01 00 00 0c f6 d8  3a dc 83 38 14 18 ff ff  |.8......:..8....|
00000010  ff ff 83 38 14 18 00 00  01 00 6a ff 00 02 ff ff  |...8......j.....|
00000020  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000030  ff ff ff ff 33 0d 02 00  33 00 00 00 00 00 ff ff  |....3...3.......|
00000040  ff ff ff ff 20 ff ff ff  ff ff ff ff 0a 0a 0a 0a  |.... ...........|
00000050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000060  06 06 06 06 06 06 06 06  06 06 06 06 06 06 05 05  |................|
00000070  05 05 05 05 05 05 05 05  05 05 05 05 06 06 06 06  |................|
00000080  06 06 06 06 06 06 06 06  06 06 00 00 00 00 00 00  |................|
00000090  00 00 00 00 ff ff 13 13  12 12 11 10 10 10 10 10  |................|
000000a0  10 10 13 13 13 13 12 12  12 11 11 11 11 11 11 11  |................|
000000b0  11 11 12 12 12 12 12 12  12 ff 00 00 00 00 00 00  |................|
000000c0  00 00 00 00 00 00 00 00  00 00 13 13 13 13 13 13  |................|
000000d0  13 13 13 13 13 13 0f 0f  0f 0f 0f 0f 0f 0f 0f 0f  |................|
000000e0  0f 0f 0f 0f 0f 0f 0d 0d  0d 0d 0d 0d 0d ff 00 00  |................|
000000f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 14 14  |................|
00000100  14 14 14 14 14 14 14 14  14 14 11 11 11 11 11 11  |................|
00000110  11 11 11 11 10 10 0f 0f  0e 0e 0f 0f 10 10 11 11  |................|
00000120  12 ff 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000130  00 00 00 00 da c2 b2 8a  81 62 42 32 22 01 00 00  |.........bB2"...|
00000140  66 ac 68 06 ac 88 c6 8a  68 ac 88 06 ac 68 06 00  |f.h.....h....h..|
00000150  00 ac 68 06 ac 88 c6 8a  68 ac 88 06 ac 68 06 00  |..h.....h....h..|
00000160  00 99 69 06 66 44 64 46  44 66 44 04 66 44 04 00  |..i.fDdFDfD.fD..|
00000170  66 99 69 06 66 44 64 46  44 66 44 04 66 44 04 00  |f.i.fDdFDfD.fD..|
00000180  00 00 00 00 00 00 00 00  00 00 00 00 00 00 ff ff  |................|
00000190  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001a0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001b0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001c0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001d0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001e0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001f0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000200
```

```
$ hexdump -v -n 512 -C './mtddump/mtd3'
00000000  92 53 03 01 00 0c f6 d8  3a d8 92 53 14 18 01 80  |.S......:..S....|
00000010  00 00 92 53 14 18 00 00  01 00 6a ff 13 02 ff ff  |...S......j.....|
00000020  ff ff c1 ff 01 92 ff ff  ff ff ff ff ff ff ff ff  |................|
00000030  8e 75 01 43 22 ff 27 00  ff ff 0e 01 ff ff d9 fa  |.u.C".'.........|
00000040  cc 88 ff ff 0a ff 00 00  00 00 00 00 00 00 ff ff  |................|
00000050  ff ff 11 11 11 11 11 11  12 12 12 12 12 12 12 12  |................|
00000060  13 13 13 14 14 14 15 15  15 16 16 16 17 17 64 22  |..............d"|
00000070  f1 00 33 11 00 00 62 de  00 00 00 00 00 00 00 00  |..3...b.........|
00000080  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000090  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000000a0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000000b0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000000c0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000000d0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff 66 66  |..............ff|
000000e0  99 99 99 66 66 66 44 44  66 66 44 44 66 66 44 44  |...fffDDffDDffDD|
000000f0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000100  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000110  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000120  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000130  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000140  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000150  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000160  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000170  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000180  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000190  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001a0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001b0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001c0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001d0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001e0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
000001f0  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000200
```

Could you have a look at your flash dumps if you could find the correct position of the EEPROM.